--- a/src/portals/Screencopy.cpp	2026-05-10 21:34:36.401190090 -0400
+++ b/src/portals/Screencopy.cpp	2026-05-10 21:34:36.401871972 -0400
@@ -474,6 +474,7 @@
                 sharingData.status = FRAME_NONE;
                 if (sharingData.copyRetries++ < MAX_RETRIES) {
                     Debug::log(LOG, "[sc] Retrying screencopy ({}/{})", sharingData.copyRetries, MAX_RETRIES);
+                    PSTREAM->dmaBufRetries = 0;
                     g_pPortalManager->m_sPortals.screencopy->m_pPipewire->updateStreamParam(PSTREAM);
                     g_pPortalManager->m_sPortals.screencopy->queueNextShareFrame(this);
                 }
@@ -589,6 +590,7 @@
                 sharingData.status = FRAME_NONE;
                 if (sharingData.copyRetries++ < MAX_RETRIES) {
                     Debug::log(LOG, "[sc] Retrying screencopy ({}/{})", sharingData.copyRetries, MAX_RETRIES);
+                    PSTREAM->dmaBufRetries = 0;
                     g_pPortalManager->m_sPortals.screencopy->m_pPipewire->updateStreamParam(PSTREAM);
                     g_pPortalManager->m_sPortals.screencopy->queueNextShareFrame(this);
                 }
